### ISSUE TITLE
try updating the 1.18.1 version/ like the loot bioems and stuff like in 1.17

### ISSUE THREAD

**Issue by Diljanshala2012** [xpple/SeedMapper#42]
add it if you can

**Comment by sbgigapsec** [comment]
that already wip dude

**Comment by sbgigapsec** [comment]
Read the disclaimer

**Comment by D-a-r-n-o-l-d** [comment]
> > try cubiomes
> 
> whats that + the loot and stuff doesnt work i tired the highlight and it doesnt work

[cubiomes](https://github.com/Cubitect/cubiomes) is a C library that mimics the minecraft world generation and is fast as hell.
This can be used in a similar way seedmapper or [ChunkBase](https://www.chunkbase.com/apps/seed-map)

There's also a GUI version that yall can use while we wait for the update called [cubiomes-viewer](https://github.com/Cubitect/cubiomes-viewer).

**Comment by vanvac** [comment]
there is this mod https://modrinth.com/mod/world-preview source: https://github.com/caeruleusDraconis/world-preview
that creates a chunk base style seed map but in game but only woks in single player with the creator saying no plans to make it work on multiplayer servers this might help with terrain/structures

**Comment by xpple** [comment]
Give me a few hours, a new release of SeedMapper supporting **all** Minecraft versions is coming.

**Comment by vanvac** [comment]
new version using cubebiomees W big well done i will report any bugs and im guessing this issue can be soon closed 🥇

**Comment by xpple** [comment]
Yeah about that... I realised that the title of this issue is 1.18+ _terrain_ generation. That is not (yet) possible with cubiomes.

**Comment by vanvac** [comment]
oh well we are getting there 👍

**Comment by NavalAce** [comment]
will you release it for versions other than 1.21.3?

**Comment by xpple** [comment]
No, however it is still possible (and encouraged!) to use SeedMapper on lower versions. To do this, download a mod like [ViaFabricPlus](https://github.com/ViaVersion/ViaFabricPlus). That mod allows you to join all servers that are on older versions of Minecraft. As always, use the latest version of both SeedMapper and ViaFabricPlus.

**Comment by NavalAce** [comment]
For now a lot of mods aren't updated yet, so I still play on 1.21.1 server with 1.21.1 modpack. It would be great if I knew a way to add your mod to it

**Comment by xpple** [comment]
I see. In that case I might add support for 1.21-1.21.3. For now, you can try unzipping the JAR and editing the `fabric.mod.json` file to allow all 1.21 versions. In particular you'd change `"minecraft": ">=1.21.3 <1.22"` to `"minecraft": "1.21.x"`. I suspect this will work, but I do not know for sure.

**Comment by NavalAce** [comment]
I just did that and it says _Unknown version "1.21.1"_ when I try to locate something via /sm:locate command

**Comment by xpple** [comment]
Ah, SeedMapper only recognises major versions now. You can force SeedMapper to use 1.21 by doing `/sm:source versioned 1.21 run <your command>`.

**Comment by xpple** [comment]
If you have further questions, let’s continue in Discord (or open a new issue) as to not stray away from the topic of this issue.
